community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/NodeValueIndexCursor.java

```diff
@@ -91,8 +96,7 @@ public void initialize( IndexDescriptor descriptor, IndexProgressor progressor,
 
     private boolean isRemoved( long reference )
     {
-        return (changes != null && changes.isRemoved( reference )) ||
-               (read.hasTxStateWithChanges() && read.txState().addedAndRemovedNodes().isRemoved( reference ));
+        return removed.contains( reference );
```

Comment:
Is it safe to cache this only at query-time?

Reply:
Hmm, not sure I know what you mean... could you explain why it would/wouldn't be safe?